shadows/framework/src/main/java/org/robolectric/shadows/ShadowBitmapFactory.java

```diff
@@ -165,6 +176,7 @@ protected static Bitmap decodeStream(
     } catch (IOException e) {
       // ignore
     }
+    initColorArray(is, ninePatchChunk, shadowBitmap);
```

Comment:
In some situations `needsReset` needs to be false, e.g:
```
  @Test
  public void readWriteAndRereadPngFile() throws IOException {
    InputStream inputStream = resources.getAssets().open("robolectric.png");
    Bitmap bitmap = BitmapFactory.decodeStream(inputStream);
    File tmp = Files.createTempFile("BitmapTest", null).toFile();
    try (FileOutputStream stream = new FileOutputStream(tmp) ) {
      bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
    }
    FileInputStream fileInputStream = new FileInputStream(tmp);
    Bitmap bitmap2 = BitmapFactory.decodeStream(fileInputStream);
    assertThat(bitmap.sameAs(bitmap2)).isTrue();
  }
```

Reply:
Actually, seems like calling mark(0) before initColorArray may fix it. I'm guessing that the mark gets adjusted when reading the image size.

```java
    is.mark(0);
   initColorArray(is, ninePatchChunk, shadowBitmap);
```

Reply:
The log and code shows me that the `FileInputStream` in OpenJDK doesn't support `mark/reset`. I will try to find a method to pre-fetch the content of stream that doesn't support `mark/reset`, or wrap it to another stream that supports `mark/reset`. It maybe needs some time, you could merge other PR about `Bitmap`, and I will rebase this PR on master. I also want to test my PR based on more stable test suites. Thanks your patience to help to debug and provide method for it.

Reply:
If the `InputStream` supports `mark/reset`, we can use it inner `initColorArray` directly. If it doesn't support, we can convert it to byte array stream, and convert saved byte array stream to different input stream for `getImageSizeFromStream` and `initColorArray`. I will try it later.